### PR TITLE
Added types for the json representation of Java Project

### LINKED ISSUE #22: Create Typescript types for JavaProject and its parts

### PR DESCRIPTION
**Pull request by Dara2004**
closes #22

### CODE REVIEW

File: client/src/JavaProjectTypes.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      name: string;
+      accessModifier: JavaAccessModifier;
+      parameters: JavaParametersMap;
+      returnType: string;
+    }
+  | {
+      name: string;
+      accessModifier: JavaAccessModifier;
+      parameters: JavaParametersMap;
+      returnType: null; //for constructor
```
Comment (algobot76) [inline client/src/JavaProjectTypes.ts:39]: I think `returnType` should be removed as a constructor returns nothing, not even `null`. ` I don't think it is accurate to classify a constructor as a method. We should create a separate type `JavaConstructor` for it.

File: client/src/JavaProjectTypes.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  accessModifier: JavaAccessModifier;
+  lineCount: number;
+  imports: string[];
+  fields: JavaFieldsMap;
+  methods: JavaMethod[];
+  constructors: JavaMethod[];
+};
+
+export type JavaPackage = {
+  package_name: string;
```
Comment (algobot76) [inline client/src/JavaProjectTypes.ts:54]: Personally, I think `package_name` is a bit verbose and `name` should be enough.

File: client/src/JavaProjectTypes.ts (hunk 1 above)
Comment (Dara2004) [reply client/src/JavaProjectTypes.ts:39]: I just followed the readme. should I change the readme as well?

File: client/src/JavaProjectTypes.ts (hunk 2 above)
Comment (Dara2004) [reply client/src/JavaProjectTypes.ts:54]: package_name is also from the readme. I can change as well if you want.

File: client/src/JavaProjectTypes.ts (hunk 1 above)
Comment (algobot76) [reply client/src/JavaProjectTypes.ts:39]: I've updated README in #26

**Comment by ZhiHanL** [comment]
We also need a JavaFile

**Comment by Dara2004** [comment]
added JavaFile and updated README with minor changes

**Comment by john-zou** [comment]
Merged into master with #69

**Pull request closed without merging**